sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdServiceEndpoint.java

```diff
@@ -185,7 +187,17 @@ public RntbdRequestRecord request(final RntbdRequestArgs args) {
 
         this.throwIfClosed();
 
-        this.concurrentRequests.incrementAndGet();
+        int concurrentRequestSnapshot = this.concurrentRequests.incrementAndGet();
+
+        if (concurrentRequestSnapshot > this.maxConcurrentRequests) {
+            return FailFastRntbdRequestRecord.createAndFailFast(
```

Comment:
Do upstream retries fail these or retry? (not a blocker now)

Reply:
No retry - that is why we chose this exception vs. GoneException. We don't want retries in this scenario